client/src/components/mainPage/mainPage.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import Header from '../header/header.component';
+import createFooter from '../footer/footer.component';
+import MoviesList from '../moviesList/moviesList.component';
+import { title, releaseDate, rating } from '../../consts';
+import movies from '../../mock-data';
+import createSortBar from '../sortBar/sortBar.component';
+import { filterByTitle, filterByGenre, sortByDate, sortByRating } from '../../utils/filterMethods';
+import styles from './mainPage.css';
+
+class MainPage extends React.Component {
+  constructor(props) {
```

Comment:
Consider removing constructor() and defining state as class property.
Same goes to binding class methods, it is redundant.
https://hackernoon.com/the-constructor-is-dead-long-live-the-constructor-c10871bea599
https://github.com/tc39/proposal-class-fields